test/framework/deployment_helpers.go

```diff
@@ -65,7 +65,13 @@ func WaitForDeploymentsAvailable(ctx context.Context, input WaitForDeploymentsAv
 		}
 		return false
 
-	}, intervals...).Should(BeTrue(), "Deployment %s/%s failed to get status.Available = True condition", input.Deployment.GetNamespace(), input.Deployment.GetName())
+	}, intervals...).Should(BeTrue(), func() string { return DescribeFailedDeployment(input, deployment) })
+}
+
+// DescribeFailedDeployment returns detailed output to help debug a deployment failure in e2e.
+func DescribeFailedDeployment(input WaitForDeploymentsAvailableInput, deployment *appsv1.Deployment) string {
+	return fmt.Sprintf("Deployment %s/%s failed to get status.Available = True condition\nService:\n%s\n",
```

Comment:
nit. 
It'd be good to add a `deployment==nil` check here, getting the deployment may be failing.